### PR TITLE
Added a benchmarking module 

### PR DESCRIPTION
**Pull request by franknoe**
for testing runtimes of hidden variable operations during development.

### CODE REVIEW

File: bhmm/tests/benchmark_hidden.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    def run_viterbi(self, i, out):
+        vpath = None
+        hidden.set_implementation(self.kernel)
+        time1 = time.time()
+        for k in range(self.nrep):
+            vpath = hidden.viterbi(self.A[i], self.pobs[i], self.pi[i])
+        # compare
+        time2 = time.time()
+        d = (time2-time1)/(1.0*self.nrep)
+        return (vpath, d)
```
Comment (jchodera) [inline bhmm/tests/benchmark_hidden.py:141]: These returns only return the _last_ result.  Did you want to append all the replicates instead?

Comment (franknoe) [inline bhmm/tests/benchmark_hidden.py:141]: No, they are identical.

Am 13/12/15 um 17:42 schrieb John Chodera:

> In bhmm/tests/benchmark_hidden.py 
> https://github.com/bhmm/bhmm/pull/26#discussion_r47446997:
> 
> > -            C = hidden.transition_counts(self.alpha[i], self.beta[i], self.A[i], self.pobs[i], out=out)
> > -        # compare
> > -        time2 = time.time()
> > -        d = (time2-time1)/(1.0*self.nrep)
> > -        return (C, d)
> >   +
> > -    def run_viterbi(self, i, out):
> > -        vpath = None
> > -        hidden.set_implementation(self.kernel)
> > -        time1 = time.time()
> > -        for k in range(self.nrep):
> > -            vpath = hidden.viterbi(self.A[i], self.pobs[i], self.pi[i])
> > -        # compare
> > -        time2 = time.time()
> > -        d = (time2-time1)/(1.0*self.nrep)
> > -        return (vpath, d)
> 
> These returns only return the /last/ result. Did you want to append 
> all the replicates instead?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/bhmm/bhmm/pull/26/files#r47446997.

## 

---

Prof. Dr. Frank Noe
Head of Computational Molecular Biology group
Freie Universitaet Berlin

Phone: (+49) (0)30 838 75354
Web:   research.franknoe.de

## Mail:  Arnimallee 6, 14195 Berlin, Germany

Comment (jchodera) [inline bhmm/tests/benchmark_hidden.py:141]: Ah, these are all deterministic computations!

Comment (franknoe) [inline bhmm/tests/benchmark_hidden.py:141]: yup

Am 13/12/15 um 17:46 schrieb John Chodera:

> In bhmm/tests/benchmark_hidden.py 
> https://github.com/bhmm/bhmm/pull/26#discussion_r47447063:
> 
> > -            C = hidden.transition_counts(self.alpha[i], self.beta[i], self.A[i], self.pobs[i], out=out)
> > -        # compare
> > -        time2 = time.time()
> > -        d = (time2-time1)/(1.0*self.nrep)
> > -        return (C, d)
> >   +
> > -    def run_viterbi(self, i, out):
> > -        vpath = None
> > -        hidden.set_implementation(self.kernel)
> > -        time1 = time.time()
> > -        for k in range(self.nrep):
> > -            vpath = hidden.viterbi(self.A[i], self.pobs[i], self.pi[i])
> > -        # compare
> > -        time2 = time.time()
> > -        d = (time2-time1)/(1.0*self.nrep)
> > -        return (vpath, d)
> 
> Ah, these are all deterministic computations!
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/bhmm/bhmm/pull/26/files#r47447063.

## 

---

Prof. Dr. Frank Noe
Head of Computational Molecular Biology group
Freie Universitaet Berlin

Phone: (+49) (0)30 838 75354
Web:   research.franknoe.de

## Mail:  Arnimallee 6, 14195 Berlin, Germany

**Pull request merged**